### DISCUSSION TITLE
Extents for Entities

### DISCUSSION THREAD

**Discussion by jameswilliambell** [DomCR/ACadSharp#205] [Q&A] [answered]
Is there any way to get the extents (bounding box) for an attribute (and/or a block). i.e. the MinX, MinY, MaxX, and MaxY values

**Comment by DJGosnell** [comment, marked as answer]
Right now you have to loop through all the entities and calculate that yourself.  You will also need to account for blocks and inserts, including rotation & scale, to get the full extents.  When I've needed to calculate the bounds in the past, especially multiple times, I've used QuadTrees to query the objects extents.

https://stackoverflow.com/questions/41946007/efficient-and-well-explained-implementation-of-a-quadtree-for-2d-collision-det
